sdk/identity/azure-identity/test/ut/credential_test_helper.cpp

```diff
@@ -45,47 +32,21 @@ class TestTransport final : public Azure::Core::Http::HttpTransport {
 
 using namespace Azure::Identity::Test::_detail;
 
-bool const CredentialTestHelper::EnvironmentOverride::IsEnvironmentAvailable(
-#if !defined(WINAPI_PARTITION_DESKTOP) \
-    || WINAPI_PARTITION_DESKTOP // See azure/core/platform.hpp for explanation.
-    true
-#else
-    false
-#endif
-);
-
 void CredentialTestHelper::EnvironmentOverride::SetVariables(
     std::map<std::string, std::string> const& vars)
 {
-#if !defined(WINAPI_PARTITION_DESKTOP) \
-    || WINAPI_PARTITION_DESKTOP // See azure/core/platform.hpp for explanation.
   for (auto var : vars)
   {
-    auto const& name = var.first;
-    auto const& value = var.second;
-
-#if defined(_MSC_VER)
-    static_cast<void>(_putenv((name + "=" + value).c_str()));
-#else
-    if (value.empty())
-    {
-      static_cast<void>(unsetenv(name.c_str()));
-    }
-    else
-    {
-      static_cast<void>(setenv(name.c_str(), value.c_str(), 1));
-    }
-#endif
+    Azure::Core::_internal::Environment::SetVariable(var.first.c_str(), var.second.c_str());
   }
-#endif
 }
 
 CredentialTestHelper::EnvironmentOverride::EnvironmentOverride(
```

Comment:
not cool.  
What happen we want to run tests in parallel?  The only environment owner should only be the user.

We should go in a different direction if we need env override. For example, adding support for `.env` file.  So if there's a .env file next to the test exe, the values on that file would override the ones set on the System env, but only within the test app. Never directly to the system env.

Reply:
btw, @LarryOsterman was looking for .env support and I think he mentioned he ended up adding it

Reply:
The variables you would set or modify would only look so for the current process or its children. I.e. practically they will have effect while the unit test exe runs. But even there in the test code we have a RAII helper that remembers the previous value (or lack of) before the testcase, sets it to a new one, and then restores - if either test case exits normally or exception is thrown.
So, if you run tests in parallel processes, each one of them may change env variables all it wants, or it may change `APPDATA`, `PATH` and so on - user will see no effect.